src/liger_kernel/transformers/monkey_patch.py

```diff
@@ -141,20 +141,28 @@ def apply_liger_kernel_to_gemma(
     assert not (
         cross_entropy and fused_linear_cross_entropy
     ), "cross_entropy and fused_linear_cross_entropy cannot both be True."
-    
+
     from transformers.models.gemma import modeling_gemma
+    from transformers.models.gemma2 import modeling_gemma2
 
     if rope:
         modeling_gemma.apply_rotary_pos_emb = liger_rotary_pos_emb
+        modeling_gemma2.apply_rotary_pos_emb = liger_rotary_pos_emb
     if rms_norm:
         # https://github.com/huggingface/transformers/blob/v4.44.2/src/transformers/models/gemma/modeling_gemma.py#L109
         modeling_gemma.GemmaRMSNorm = partial(LigerRMSNorm, offset=1.0, init_fn="zeros")
+        modeling_gemma2.Gemma2RMSNorm = partial(
+            LigerRMSNorm, offset=1.0, init_fn="zeros"
+        )
     if cross_entropy:
         modeling_gemma.CrossEntropyLoss = LigerCrossEntropyLoss
+        modeling_gemma2.CrossEntropyLoss = LigerCrossEntropyLoss
     if geglu:
         modeling_gemma.GemmaMLP = LigerGEGLUMLP
+        modeling_gemma2.Gemma2MLP = LigerGEGLUMLP
     if fused_linear_cross_entropy:
         modeling_gemma.GemmaForCausalLM.forward = gemma_lce_forward
+        modeling_gemma2.Gemma2ForCausalLM.forward = gemma_lce_forward
```

Comment:
It looks like the same recently added lce_forward is being used for both Gemma and Gemma2 (https://github.com/linkedin/Liger-Kernel/pull/111).

It appears that there is a slight difference in the `forward` between Gemma and Gemma2 in the modeling code in transformers, specifically regarding logit softcapping (https://github.com/huggingface/transformers/blob/main/src/transformers/models/gemma2/modeling_gemma2.py#L1054-L1057), which doesn't seem to be accounted for in the new lce code as far as I can tell. Would this potentially lead to incompatibility?

Reply:
yeah exactly, just figured out we need to add softcapping :/ i'll have a follow up PR to clarify that right now this only works for gamma 1 and 1.1 and we'll work on changing the kernel correspondingly asap